finesse/hardware/plugins/time/__init__.py

```diff
@@ -1 +1,23 @@
 """This module contains interfaces for querying time sources."""
+
+from datetime import datetime
+
+from finesse.hardware.manage_devices import get_device_instance
+from finesse.hardware.plugins.time.time_base import TimeBase
+
+
+def get_time_instance() -> TimeBase | None:
+    """Get the instance of the current time source if connected or None."""
+    return get_device_instance(TimeBase)
+
+
+def get_time() -> datetime:
+    """Read the current time, with optional time offset.
+
+    If the time device is not connected or the operation fails, local time is returned.
+    """
+    timestamp = datetime.now().timestamp()
+    if dev := get_time_instance():
+        timestamp += dev.get_time_offset()
+
+    return datetime.fromtimestamp(timestamp)
```

Comment:
Instead of converting back and forth between epoch, you could do:

```suggestion
    time = datetime.now()
    if dev := get_time_instance():
        return time + timedelta(seconds=dev.get_time_offset())

    return time
```

(You need `from datetime import timedelta`)